search/retrieval-augmented-generation/rag_with_dual_llms/src/vertex_rag_demo_dual_llms_with_judge.py

```diff
@@ -0,0 +1,1177 @@
+            # http_options=types.HttpOptions(api_version=DEFAULT_GEMINI_API_VERSION) # Uncomment if specific version needed
+        )
+
+
+def _load_prompt_template(filename: str) -> Optional[str]:
+    """Loads a prompt template file from the PROMPT_FOLDER."""
+    filepath = os.path.join(PROMPT_FOLDER, filename)
+    return load_text_file(filepath)
+
+
+def _apply_system_prompt(prompt_text: str) -> str:
+    """Prepends the system prompt from session state if available."""
+    system_instruction = st.session_state.get("system_prompt", "").strip()
+    if system_instruction:
+        return f"{system_instruction}\n\n{prompt_text}"
+    return prompt_text
+
+
+def get_rephraser_prompt(query: str) -> Optional[str]:
+    """
+    Loads and formats the rephraser prompt.
+
+    Args:
+        query: The user's original query.
+
+    Returns:
+        The formatted prompt string, or None if loading fails.
+    """
+    prompt_template = _load_prompt_template(REPHRASER_PROMPT_FILE)
+
+    try:
+        formatted_prompt = """Now, please rephrase the following customer query:
+   
+{query}
+"""
+        formatted_prompt = prompt_template + formatted_prompt.format(query=query)
+        return _apply_system_prompt(formatted_prompt)
+    except KeyError as e:
+        logging.error(f"Error formatting rephraser prompt: Missing key {e}")
+        st.error("Internal error: Could not format the rephrasing prompt.")
+        return None
+
+
+def get_summarizer_prompt(documents: List[str], query: str) -> Optional[str]:
+    """
+    Loads and formats the summarizer prompt with retrieved documents.
+
+    Args:
+        documents: A list of context documents.
+        query: The user's query (potentially rephrased).
+
+    Returns:
+        The formatted prompt string, or None if loading fails.
+    """
+    prompt_template = _load_prompt_template(SUMMARIZER_PROMPT_FILE)
+
+    document_vars = {}
+    if documents:
+        for i, doc in enumerate(documents):
+            document_vars[f"Text_of_Document_{i + 1}"] = doc
+    else:
+        # Provide placeholder if no documents are found
+         document_vars["Text_of_Document_1"] = "No relevant documents found."
+
+
+    # Prepare format arguments - ensure all expected keys are present
+    # Assuming the template expects up to 3 docs, provide placeholders
+    format_args = {"query": query}
+    for i in range(1, 4): # Adjust range if template expects different number
+        key = f"Text_of_Document_{i}"
+        format_args[key] = document_vars.get(key, "No document provided for this slot.")
+
+    formatted_prompt = """
+Now it's your turn! Here is the query and relevant documents:
+    Customer Search Query: {query}
+    
+    Document Texts:
+    [Start of Document 1]
+    {Text_of_Document_1}
+    [End of Document 1]
+    
+    [Start of Document 2]
+    {Text_of_Document_2}
+    [End of Document 2]
+    
+    [Start of Document 3]
+    {Text_of_Document_3}
+    [End of Document 3]    
+    """
```

Comment:
Looks like this is a duplicate from your other prompt files. Can you edit this to use your prompt files?

Reply:
This is actually the code that over-rides whatever is in the .txt file since users may write all kinds of summarizer Prompts. This will ensure that their prompts will still work with our retrieval strategies.